fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/simplification/Simplification.java

```diff
@@ -444,58 +491,32 @@ public static <ARGUMENT> NonnullPair<QueryPredicate, List<QueryPlanConstraint>>
-                                                                                                                                                                      @Nonnull BASE current,
-                                                                                                                                                                      @Nonnull final Map<BASE, NonnullPair<BASE, List<ELEMENT>>> resultsMap,
-                                                                                                                                                                      @Nonnull final AbstractRuleSet<CALL, BASE> ruleSet,
-                                                                                                                                                                      @Nonnull final RuleCallCreator<NonnullPair<BASE, List<ELEMENT>>, CALL, BASE> ruleCallCreator) {
+    private static <CALL extends AbstractRuleCall<BASE, CALL, BASE>, BASE extends TreeLike<BASE>> BASE simplifyWithReExploration(@Nonnull final BASE root,
+                                                                                                                                 @Nonnull BASE current,
+                                                                                                                                 @Nonnull final Map<BASE, QueryPlanConstraint> constraintsMap,
+                                                                                                                                 @Nonnull final AbstractRuleSet<CALL, BASE> ruleSet,
+                                                                                                                                 @Nonnull final RuleCallCreator<BASE, CALL, BASE> ruleCallCreator) {
         final var isRoot = root == current;
         ExecutionResult<BASE> executionResult;
         do {
             final var simplifiedChildren = Lists.<BASE>newArrayList();
             for (final var child : current.getChildren()) {
-                simplifiedChildren.add(simplifyWithReExploration(isRoot ? current : root, child, resultsMap, ruleSet, ruleCallCreator));
-            }
-
-            final var computedCurrent = computeCurrent(current, simplifiedChildren);
-            if (computedCurrent != current) {
-                if (resultsMap.containsKey(current)) {
-                    resultsMap.put(computedCurrent, NonnullPair.of(computedCurrent, resultsMap.get(current).getRight()));
-                } else {
-                    final var computedCurrentElements = collectChildrenElements(resultsMap, simplifiedChildren);
-                    final var computedCurrentPair = NonnullPair.of(computedCurrent, computedCurrentElements);
-                    resultsMap.put(computedCurrent, computedCurrentPair);
-                }
+                simplifiedChildren.add(simplifyWithReExploration(isRoot ? current : root, child, constraintsMap, ruleSet, ruleCallCreator));
```

Comment:
Is this correct? `current` can change in line `507`, however it is not matched against root as in `499`.